### PR TITLE
[ISM][#18651] TierPrice save float percentage value [2.3]

### PR DESCRIPTION
**Pull request by ghost**
### Description
Problem:
After save tierprice for product, `discount` field lost value after comma.
Fix:
Updated getPercentage method in TierPrices:
- Type conversation changed: int -> float
- Added strict types
- Database has remained unchanged since the percentage_value is a decimal field

### Fixed Issues
1. magento/magento2#18651: Tierprice can't save float percentage value

### Manual testing scenarios
1. Add new product
2. Add new tierprice with `discount` value "2.5"
3. Save the product
4. Value "2.5" is saved

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @SikailoISM. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by rogyar** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rogyar. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rogyar, here is your [new Magento instance](http://34.228.235.121/pr-19140/).
Admin access: [http://34.228.235.121/pr-19140//admin](http://34.228.235.121/pr-19140//admin)
Login: `admin` Password: `123123q`

**Comment by magento-engcom-team** [comment]
@SikailoISM thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Review by orlangur — CHANGES_REQUESTED** [review]
https://github.com/magento/magento2/pull/19030#discussion_r231359724 CR note is missed.

**Comment by orlangur** [comment]
@SikailoISM please apply necessary changes.

File: app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php [hunk 1]
```diff
@@ -135,13 +135,11 @@ private function getAdditionalFields(array $objectArray): array
      * Check whether price has percentage value.
      *
      * @param array $priceRow
-     * @return int|null
+     * @return float
      */
-    private function getPercentage(array $priceRow): ?int
+    private function getPercentage(array $priceRow): float
```
Comment (orlangur) [inline app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php:140]: Will such type hinting work when there is `int`?

Comment (ghost) [reply app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php:140]: Yes, of course.

Comment (swnsma) [reply app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php:140]: image

Comment (swnsma) [reply app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php:140]: Just checked :)

Comment (orlangur) [reply app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php:140]: @swnsma nice idea ;) such signature will convert to `float` though which is not desired: https://3v4l.org/eE28s

Please remove return type and make it possible to pass `int` unchanged.

**Pull request closed without merging**